### PR TITLE
KV configuration block

### LINKED ISSUE #3505: Allow configuration block for any KV implementation
We currently use a single dsn string to fit them all. That doesn't serve us anymore because:
1. Not all KV implementations configuration are easily mapped to a single string.
2. There are cases where we need more than 1, for example during migration.

### PR DESCRIPTION
**Pull request by itaiad200**
closes #3505

### CODE REVIEW

File: pkg/kv/params/database.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package params
+
+type KV struct {
+	Postgres *Postgres
+	DynamoDB *DynamoDB
+}
```
Comment (itai-david) [inline pkg/kv/params/database.go:6]: Very elegant :+1:

File: pkg/kv/store.go [hunk 2]
```diff
@@ -146,14 +148,14 @@ func UnregisterAllDrivers() {
 
 // Open lookup driver with 'name' and return Store based on 'dsn' (data source name).
 // Failed with ErrUnknownDriver in case 'name' is not registered
-func Open(ctx context.Context, name, dsn string) (Store, error) {
+func Open(ctx context.Context, name string, params kvparams.KV) (Store, error) {
 	driversMu.RLock()
 	d, ok := drivers[name]
 	driversMu.RUnlock()
```
Comment (itai-david) [inline pkg/kv/store.go:154]: I know it's not part of your PR, but it caught my eye
```suggestion
	defer driversMu.RUnlock()
```

File: cmd/lakefs/cmd/import.go [hunk 3]
```diff
@@ -125,7 +125,8 @@ func runImport(cmd *cobra.Command, args []string) (statusCode int) {
 	var idGen actions.IDGenerator
 	var actionsStore actions.Store
 	if dbParams.KVEnabled {
-		kvStore, err := kv.Open(ctx, dbParams.Type, dbParams.ConnectionString)
+		kvparams := cfg.GetKVParams()
+		kvStore, err := kv.Open(ctx, dbParams.Type, kvparams)
```
Comment (itai-david) [inline cmd/lakefs/cmd/import.go:129]: Just a thought:
How about the option to send the entire `cfg` to `kv.Open` and it perfrom the `cfg.GetKVParams()`?
On a logical level it means that KV is "aware" of the entire `config` package and that it (`cfg`) has a designated section (`kvparams`) for its ('KV') use, which makes sense, and that the `cmd` package, does not need to be aware of `kvparams` which also makes sense (especially if and when `kvparams` will change)

**Review by itai-david — APPROVED** [review]
Looks great - clear and simple.
One thought regarding the right location to perform the `kvparams` extraction from `cfg`. That's just an idea and up to you whether to consider it or not. Not a change request

File: cmd/lakefs/cmd/import.go (hunk 3 above)
Comment (itaiad200) [reply cmd/lakefs/cmd/import.go:129]: I think it's better that the wiring happens and a single place. Also `kv` doesn't have a dependency on the `config` pkg, it does have a dependency on its inner `params` package. That's the same pattern the `db` package uses and without clear advantages to any solution, I rather keep it consistent.

**Pull request merged**